src/validators/validators.ts

```diff
@@ -0,0 +1,15 @@
+export function validator(
+  validationFunction: ValidatorFunction,
+  message: String
+): Validator {
+  return (keyPath, value) => {
+    if (!validationFunction(value)) {
+      throw Error(
+        `Invalid configuration value ${value} for ${keyPath}: ${message}`
+      );
+    }
+  };
+}
+
+type ValidatorFunction = (value: any) => boolean;
```

Comment:
Is your plan to have esm-module-config export some premade validator functions for strings, numbers, objects, arrays, etc?

Reply:
Yep!